android-sdk/README.md

```diff
@@ -0,0 +1,224 @@
+
+For example, if a KeyCloak and fh-sync-server service were created and bound, the developer was responsible for retrieving tokens from keycloak and configuring fh-sync-server to consume those tokens.  However, fh-sync-server was unaware of tokens beyond the fact that they were to be included in headers of http requests.  If a token expired fh-sync-server would return to the user a synchronization error and the developer would have had to code a system for parsing that error, understanding that it was a token timeout, refreshing the token, and restarting the sync operation.  In the mobile core world the Core SDK would understand that the sync service module connection error was a token error and invoke the keycloak module to resolve it automatically.
+
+## Proposed solution
+
+### Core SDK
+
+The Core SDK will be the package that all service modules inherit from.  It will be responsible for managing network requests, threading, logging, error handling and resolution, loading configuration for mobile-core services, and other cross cutting concerns.  Modules will interface with the core SDK by implementing the necessary interfaces provided and annotating the implementations so that the Core SDK will be able to correctly manage them as well.  The Core SDK will also allow some of its internals to be exposed through interfaces for purposes such as testing, bespoke integrations, and management of additional resources.
+
+The Core SDK will provide a request lifecycle that the Psuedocode section will demonstrate.
+
+### Service Modules
+
+Service modules will each represent a singular service that is supportd by mobile-core.  We will use, where available and practical, existing libraries for services and in those cases the modules will provide the necessary implementation of Core APIs to make them compatable with the Core SDK lifecycles and features.  The first Services we are looking at integrations will be Keycloak, fh-sync-service, and a RESTful API service module.
+
+### Implementation notes
+
+Public APIs should always expose native Java Objects and not stringified JSON or JSON Objects.
+
+### Use cases and psuedo code examples
+
+#### Starting up Mobile Core
+MobileCore is the "main" class for the SDK of the Core SDK.  Users instanciate it by placing a client configuration file in their assets directory, configuring MobileCore.Builder, and then calling its build method.
```

Comment:
I guess it has to be called from the app's `Application` class? Would it be worth mentioning that here? Moreover, if the building is not called at all, would an exception be triggered or something?

Reply:
No not really.  It is up to each application's needs.  For instance someone may be building the MobileCore in a ApplicationScope dagger provider.

If building isn't called at all that means either someone is managing their modules manually or is going to have a NPE when they reference a module; either way an exception is not appropriate because they didn't use the builder.